src/lib/statistics/statistics_objects/top_k_uniform_distribution_histogram.cpp

```diff
@@ -0,0 +1,146 @@
+#include "top_k_uniform_distribution_histogram.hpp"
+
+#include <cmath>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+#include <iterator>
+#include <algorithm>
+
+#include <tsl/robin_map.h>  // NOLINT
+
+#include "generic_histogram.hpp"
+#include "resolve_type.hpp"
+#include "storage/segment_iterate.hpp"
+
+#include "expression/evaluation/like_matcher.hpp"
+#include "generic_histogram.hpp"
+#include "generic_histogram_builder.hpp"
+#include "lossy_cast.hpp"
+#include "resolve_type.hpp"
+#include "statistics/statistics_objects/abstract_statistics_object.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/segment_iterate.hpp"
+#include "abstract_histogram.hpp"
+#include "equal_distinct_count_histogram.hpp"
+#include "equal_distinct_count_histogram.cpp"
+
+
+namespace opossum {
+
+template <typename T>
+std::shared_ptr<GenericHistogram<T>> TopKUniformDistributionHistogram<T>::from_column(
+    const Table& table, const ColumnID column_id, const HistogramDomain<T>& domain) {
+
+  auto value_distribution = value_distribution_from_column(table, column_id, domain);
+
+  if (value_distribution.empty()) {
+    return nullptr;
+  }
+
+  // If the column holds less than K distinct values use the distinct count as TOP_K instead
+
+  const auto k = std::min(TOP_K_DEFAULT, value_distribution.size());
+
+  // Get the first top k values and save them into vectors
+  std::vector<T> top_k_names(k);
+  std::vector<HistogramCountType> top_k_counts(k);
+
+  // Sort values by occurrence count
+  auto sorted_count_values = value_distribution;
```

Comment:
Do we need to copy it? We could also rewrite value_distribution_from_column so that it doesn't sort and then only use one array on which we sort. - Leo

Reply:
Can we move value_distribution_from_column to abstract histogram?

Reply:
When creating bins we rely on value_distribution being sorted according to value.